db/db_test.cc

```diff
@@ -2871,12 +2871,13 @@ TEST_F(DBTest, DISABLED_FIFOCompactionWithTTLTest) {
         ASSERT_OK(Put(ToString(i * 20 + j), RandomString(&rnd, 980)));
       }
       Flush();
+      ASSERT_OK(dbfull()->TEST_WaitForCompact());
     }
-    ASSERT_OK(dbfull()->TEST_WaitForCompact());
     ASSERT_EQ(NumTableFilesAtLevel(0), 10);
 
-    // sleep for 5 seconds
-    env_->SleepForMicroseconds(5 * 1000 * 1000);
+    // sleep for 2 seconds
+    env_->SleepForMicroseconds(2 * 1000 * 1000);
```

Comment:
No, the test will not work if the the sleep is removed. This sleep is waiting for the ttl to expire. 
On line 2863, I initially set the TTL to 1 hour (arbitrary), and created all the files. On line 2885, I then change the TTL to 1 second ... but since at least 2 seconds have passed by now due to sleeping, all the files are ready for compaction/deletion because of ttl expriy. Line 2889 shows that all the files are deleted. 

`MemEnv` provides a method to mock sleep by doing a `FakeSleepForMicroseconds`. I wish I could use it to sleep longer to mimic realistic TTL values, say, like 1 day or 1 week. But I couldn't yet get FakeSleep to work even after spending considerable time and effort.